plugins/woocommerce/includes/class-wc-order.php

```diff
@@ -1721,7 +1742,8 @@ public function get_downloadable_items() {
 	 * @return bool
 	 */
 	public function needs_payment() {
-		$valid_order_statuses = apply_filters( 'woocommerce_valid_order_statuses_for_payment', array( 'pending', 'failed' ), $this );
+		// phpcs:disable WooCommerce.Commenting.CommentHooks.MissingHookComment
```

Comment:
I know it's extra work, but can we add necessary comments instead of the introduced suppressions?

Reply:
Oops! I forgot this one and some other cases. I just updated all of them adding the missing comments.